Drivers/Periphs/Src/SDCard.c

```diff
@@ -1,48 +1,74 @@
+/**
+ * @file SDCard.c
+ * @brief SD Card API
+ * 
+ * @copyright Copyright (c) 2022 UT Longhorn Racing Solar
+ * 
+ */
+
+#include "SDCard.h"
+#include "fatfs.h"
+#include "main.h"
+#include "config.h"
+#include "LED.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <stdarg.h> //for va_list var arg functions
-#include "fatfs.h"
-#include "SDCard.h"
-#include "main.h"
+#include <inttypes.h>
+
+#define SDC_ID_ENUM_TO_IDX(x)       (x-1)   // x is of type SDCardID_t
+typedef struct {
+    const char fname[15];
+    FIL file;
+    int (* sprint)(char *, size_t, void *, const char *);
+    char writebuf[SDCARD_WRITE_BUFSIZE];
```

Comment:
Do we want the buffer size for each file to be 512 or the total buffer size to be 512?

Reply:
Each file, since 512B is around the optimal chunk write size and files are written one at a time